src/web/command_control/client/components/CommandControl.tsx

```diff
@@ -1079,6 +1079,38 @@ export default class CommandControl extends React.Component {
 		this.missionHistory.push(deepcopy(mission))
 	}
 
+	/**
+	 * Used to differentiate between a state change and a task/mission change. Called from onDoneClick() in Render(). Calls updateMissionHistory() if the task of a waypoint changes. 
+	 * 
+	 * @param {MissionInterface} mission Provides current mission to test if task type has changed between the previous and latest missions
+	 * @returns {void}
+	 * 
+	 * @notes
+	 * Tests to see if there was a change in the type of task for each waypoint. If the done button was pressed in the goal settings panel,
```

Comment:
Something similar here, what do you of reducing this to the final sentence?

Reply:
Done.